streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -221,7 +239,18 @@ public StreamsConfig(Map<?, ?> props) {
     }
 
     public Map<String, Object> getConsumerConfigs(StreamThread streamThread, String groupId, String clientId) {
-        Map<String, Object> props = getBaseConsumerConfigs();
+        // filter out non-consumer configs
+        Map<String, Object> props = ConsumerConfig.filter(this.originals());
+
+        // enforce streams overrides if not specified by user
+        for (String keyName : CONSUMER_OVERRIDES.keySet()) {
+            if (!props.containsKey(keyName))
+                props.put(keyName, CONSUMER_OVERRIDES.get(keyName));
+        }
+
+        // disable auto commit ignoring any user overridden values,
+        // this is necessary for streams commit semantics
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
```

Comment:
Is ignoring the right answer? Would it be better to fail fast? If not, at least a warning should be logged.